### PR TITLE
[core][distributed] add pynccl broadcast

### PR DESCRIPTION
**Pull request by youkaichao**

### CODE REVIEW

File: tests/distributed/test_pynccl.py [hunk 1]
```diff
@@ -293,6 +293,36 @@ def test_pynccl_multiple_send_recv():
...
+        for i in range(pynccl_comm.world_size)
+    ]
+    recv_tensors[pynccl_comm.rank] = torch.ones(
+        16, 1024, 1024, dtype=torch.float32,
+        device=pynccl_comm.device) * pynccl_comm.rank
+
+    for i in range(pynccl_comm.world_size):
+        pynccl_comm.broadcast(recv_tensors[i], src=i)
+        result = recv_tensors[i].mean().cpu().item()
+        assert result == i
```
Comment (tlrmchlsmth) [inline tests/distributed/test_pynccl.py:323]: Does this work instead?
```suggestion
        assert torch.all(recv_tensors[i] == i).cpu().item()
```

**Review by tlrmchlsmth — APPROVED** [review]
looks good

I noticed that many of the tests in test_pynncl are checking the mean when it looks like they ought to be all the same -- we can fix it up in a follow-up

File: tests/distributed/test_pynccl.py (hunk 1 above)
Comment (youkaichao) [reply tests/distributed/test_pynccl.py]: yes it works, changed in [e116a89](https://github.com/vllm-project/vllm/pull/10843/commits/e116a89020b058760da84b11be4a124ebabf3b64)

Comment (youkaichao) [reply tests/distributed/test_pynccl.py]: also locally tested

**Pull request merged**